statistics/histogram.go

```diff
@@ -1108,20 +1112,23 @@ func (c *Column) IsInvalid(sctx sessionctx.Context, collPseudo bool) bool {
 		if stmtctx != nil && stmtctx.StatsLoad.Fallback {
 			return true
 		}
-		if c.Histogram.NDV > 0 && c.notNullCount() == 0 && stmtctx != nil {
+		if !c.Loaded && stmtctx != nil {
```

Comment:
How about wrapping this logic in a dedicated function like `c.IsLoaded()`?
Then we can implement it as `c.Loaded || (c.Histogram.NDV > 0 && c.notNullCount() == 0)` which is safer and easier to maintain(for example, if we want to update this logic in the future, we just need to modify one place).

Reply:
Updated.
But I'd like to think of it as insurance as we are switching from the old method to using `Loaded`. I expect to remove the method someday and use the single `Loaded` variable to check.